src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -105,8 +115,46 @@ protected internal override PinMode GetPinMode(int pinNumber)
 
         protected internal override bool IsPinModeSupported(int pinNumber, PinMode mode)
         {
-            // Libgpiod Api do not support pull up or pull down resistors for now.
-            return mode != PinMode.InputPullDown && mode != PinMode.InputPullUp;
+            switch (mode)
+            {
+                case PinMode.Input:
+                    return true;
+                case PinMode.InputPullDown:
+                case PinMode.InputPullUp:
+                    // for use the bias flags we need libgpiod version 1.5 or later
+                    IntPtr libgpiodVersionPtr = Interop.libgpiod.gpiod_version_string();
+                    string libgpiodVersionMatch = Marshal.PtrToStringAnsi(libgpiodVersionPtr);
+                    Version libgpiodVersion = new Version(libgpiodVersionMatch);
+                    bool isLibgpiod1dot5 = (libgpiodVersion.Major >= 1 && libgpiodVersion.Minor >= 5);
+
+                    // for use the bias flags we need Kernel version 5.5 or later
+                    string[] linuxVersionMatch = RuntimeInformation.OSDescription.Replace("Linux ", string.Empty).Split('.');
+                    Version linuxVersion = new Version();
```

Comment:
would it make sense to do something like: `Version linuxVersion = (linuxVersionMatch.Length > 1) ? ... : null;`
and then assign `bool isKernelLinux5dot5 = linuxVersion != null && ...;` to avoid extra allocations?